x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlTranslatorHandler.java

```diff
@@ -271,4 +284,91 @@ public static Query doTranslate(ScalarFunction f, TranslatorHandler handler) {
+            Check.isTrue(
+                constantExpression.foldable(),
+                "Line {}:{}: Comparisons against fields are not (currently) supported; offender [{}] in [ST_{}]",
+                constantExpression.sourceLocation().getLineNumber(),
+                constantExpression.sourceLocation().getColumnNumber(),
+                Expressions.name(constantExpression),
+                queryRelation
+            );
+        }
+
+        /**
+         * We should normally be using the real `wrapFunctionQuery` above, so we get the benefits of `SingleValueQuery`,
+         * but at the moment `SingleValueQuery` makes use of `SortDocValues` to determine if the results are single or multi-valued,
+         * and LeafShapeFieldData does not support `SortedBinaryDocValues getBytesValues()`.
+         * Skipping this code path entirely is a temporary workaround while separate work is being done to simplify `SingleValueQuery`
+         * to rather rely on a new method on `LeafFieldData`. This is both for the benefit of the spatial queries, as well as an
+         * improvement overall.
+         * TODO: Remove this method and call the parent method once the SingleValueQuery improvements have been made
+         */
+        public static Query wrapFunctionQuery(Expression field, Supplier<Query> querySupplier) {
+            return ExpressionTranslator.wrapIfNested(querySupplier.get(), field);
+        }
+
+        public static Query doTranslate(SpatialRelatesFunction bc, TranslatorHandler handler) {
+            if (bc.left().foldable()) {
+                checkSpatialRelatesFunction(bc.left(), bc.queryRelation());
+                return wrapFunctionQuery(bc.right(), () -> translate(bc, handler));
+            } else {
+                checkSpatialRelatesFunction(bc.right(), bc.queryRelation());
+                return wrapFunctionQuery(bc.left(), () -> translate(bc, handler));
+            }
+        }
+
+        static Query translate(SpatialRelatesFunction bc, TranslatorHandler handler) {
+            TypedAttribute attribute = bc.left().foldable() ? checkIsPushableAttribute(bc.right()) : checkIsPushableAttribute(bc.left());
+            String name = handler.nameOf(attribute);
+            Object value = valueOf(bc.left().foldable() ? bc.left() : bc.right());
+
+            if (value instanceof BytesRef bytesRef) {
+                try {
+                    Geometry shape = WellKnownBinary.fromWKB(
+                        GeometryValidator.NOOP,
+                        false,
+                        bytesRef.bytes,
+                        bytesRef.offset,
+                        bytesRef.length
+                    );
+                    return new SpatialRelatesQuery(bc.source(), name, shapeRelation(bc), shape, attribute.dataType());
+                } catch (IllegalArgumentException notWKB) {
+                    // TODO don't use exceptions to detect WKB vs WKT
```

Comment:
ouch! Don't we know here what is the format? My expectation is that the literal coming from the query is always WKT? I think this needs to be addressed before pushing it

Reply:
I will investigate which exact use case caused this, and try remove it.

Reply:
There is now a case of this in SpatialRelatesFunction too, and that is part of the implicit string conversion support, so it should go with the next update where we remove implicit conversion of strings.